staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -442,8 +439,20 @@ func (o *ExposeServiceOptions) createService() (*corev1.Service, error) {
 			case len(protocol) == 0 && len(portProtocolMap) > 0:
 				// no --protocol and we expose a multiprotocol resource
 				protocol = "TCP" // have the default so we can stay sane
-				if exposeProtocol, found := portProtocolMap[stillPortString]; found {
-					protocol = exposeProtocol
+				if exposeProtocols, found := portProtocolMap[stillPortString]; found {
+					if len(exposeProtocols) == 1 {
+						protocol = exposeProtocols[0]
+						break
+					}
+					for _, exposeProtocol := range exposeProtocols {
+						name := fmt.Sprintf("port-%d-%s", i+1, strings.ToLower(exposeProtocol))
```

Comment:
What is the reason of this name generation mechanism?

Reply:
The generation rule of this name is to be consistent with the existing rules

 The current rule is 

```
name := fmt.Sprintf("port-%d", i+1）
```

In order to solve the problem of duplication, the protocol will be added later


```
name := fmt.Sprintf("port-%d-%s", i+1, protocol)
```